mmv1/products/firebasehosting/api.yaml

```diff
@@ -0,0 +1,67 @@
+# Copyright 2018 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: FirebaseHosting
+display_name: Firebase Hosting
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://firebasehosting.googleapis.com/v1beta1/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+objects:
+  - !ruby/object:Api::Resource
+    name: 'Site'
+    min_version: beta
+    base_url: projects/{{project}}/sites
+    self_link: projects/{{project}}/sites/{{site_id}}
+    create_url: projects/{{project}}/sites?siteId={{site_id}}
+    update_verb: :PATCH
+    update_mask: true
+    description: 'A `Site` represents a Firebase Hosting site.'
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://firebase.google.com/'
+      api: 'https://firebase.google.com/docs/reference/hosting/rest/v1beta1/projects.sites'
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: 'site_id'
+        description: |
+          Required. Immutable. A globally unique identifier for the Hosting site. This identifier is
+          used to construct the Firebase-provisioned subdomains for the site, so it must also be a valid
+          domain name label.
+        input: true
+        url_param_only: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: name
+        description: |
+          Output only. The fully-qualified resource name of the Hosting site, in the
+          format: projects/PROJECT_IDENTIFIER/sites/SITE_ID PROJECT_IDENTIFIER: the
+          Firebase project's
+          [`ProjectNumber`](https://firebase.google.com/docs/projects/api/reference/rest/v1beta1/projects#FirebaseProject.FIELDS.project_number) ***(recommended)*** or its
+          [`ProjectId`](https://firebase.google.com/docs/projects/api/reference/rest/v1beta1/projects#FirebaseProject.FIELDS.project_id).
+          Learn more about using project identifiers in Google's
+          [AIP 2510 standard](https://google.aip.dev/cloud/2510).
+        output: true
+      - !ruby/object:Api::Type::String
```

Comment:
probably should be a resource ref: https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1#resourceref

now that we can create/delete web-apps properly.

Reply:
Great idea. However, this will be a cross-product reference, which I can't find an example of. I also tried "WebApp" and "Firebase.WebApp", but no magic there.
@melinath I also saw a few comments about cross-product refs such as this 4-year old one https://github.com/GoogleCloudPlatform/magic-modules/blame/ceeae1e233901a6c024f73322ee6fbd340e3d177/mmv1/products/dns/api.yaml#L193. Are they supported now?

Reply:
I'd say just leave it a String for now. ResourceRef is weird and might not behave the way we expect.